x-pack/plugins/apm/server/utils/merge_periods_timeseries.test.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { mergePeriodsTimeseries } from './merge_periods_timeseries';
+
+describe('mergePeriodsTimeseries', () => {
+  it('returns undefined when timeseries is empty', () => {
+    expect(
+      mergePeriodsTimeseries({
+        currentPeriodTimeseries: [],
+        previousPeriodTimeseries: [],
+      })
+    ).toEqual([]);
+  });
+  it('returns undefined when comparison timeseries is undefined', () => {
+    expect(
+      mergePeriodsTimeseries({
+        currentPeriodTimeseries: [],
+        previousPeriodTimeseries: undefined,
+      })
+    ).toEqual([]);
+  });
+  it('merges timeseries and comparison data', () => {
```

Comment:
> can you add a test here that merges the data when the timeseries for the current period is empty, but the previous period is not? and by empty I mean an empty array, as there are multiple reasons for that to happen

sure.

> (I do still worry about the way we are merging buckets here and depending on the fact that both searches will return the exact same amount of buckets).

The `currentPeriodTimeseries` is the one that guides the merge. 
```js
return currentPeriodTimeseries?.map(({ x, y }, index) => {
```
If the number of buckets is not in sync, we'd only show the previous periods that match the time in the current period. And the chart would show some gaps for the previous period area chart.

Reply:
It doesn't matter which one guides the merge. If the number of buckets is not the same, data will incorrectly be merged. E.g., if the current period has 30 buckets, but the previous period has 60 buckets, the values for the previous bucket will be twice as high, and it will not report data for half of the time range.

Reply:
For throughput it might not matter, as I assume that is pre-normalized. But it's really hard to figure out the scenarios where this will or will not work.

Reply:
Indeed, but I don't see an alternative here until it is fixed on Elastic charts.